controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -42,11 +51,21 @@ const InitDiscoveryTopicData = async (req, res) => {
     });
     let suggestedTopics = topicWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData[0]?.total_data;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedTopics,
-      page: page + 1
+      page: page + 1,
+      total_page: Math.ceil(totalData / limit)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The response object is correctly updated to include the total pages based on the calculated total data and provided limit. However, it would be better to handle the case where `limit` is 0 or undefined to avoid division by zero error.
```diff
-       total_page: Math.ceil(totalData / limit)
+       total_page: limit ? Math.ceil(totalData / limit) : 0
```

Overall, the changes made in this PR are mostly correct, but the SQL query needs to be revised to accurately calculate the total data count. Additionally, handling for division by zero should be added when calculating the total pages.


<!-- This is an auto-generated comment by OSS CodeRabbit -->